exla/lib/exla/defn.ex

```diff
@@ -2545,6 +2546,50 @@ defmodule EXLA.Defn do
 
   ## Cond
 
+  defp to_if(pred, on_true, on_false, %{builder: %Function{} = function} = state, cache) do
+    {pred_op, cache} = recur_operator(pred, state, cache)
+    # pred_op = to_type(pred_op, {:pred, 8})
+
+    true_ids = Tree.scope_ids(on_true)
+    false_ids = Tree.scope_ids(on_false)
+
+    cache = recur_shared_ids(on_true, false_ids, state, cache)
+    cache = recur_shared_ids(on_false, true_ids, state, cache)
+
+    out_shape = container_to_exla_shape(on_true)
+
+    in_token = get_token(cache)
+
+    result_shapes =
+      if in_token do
+        [EXLA.Shape.make_token_shape() | out_shape]
+      else
+        out_shape
+      end
+
+    [node | _] = if_results = Value.if_op(pred_op, result_shapes)
+
+    cache = to_mlir_if_branch(true, node, on_true, true_ids, state, cache)
+
+    # We need to set the outer token back in the cache so that the false branch also
+    # sees the same input token. The branch that executes will either update or return the token as is,
+    # but both branch off of the same token.
+
+    # the output token is `node` as above, if the computation does indeed contain a token.
+
+    cache =
+      to_mlir_if_branch(false, node, on_false, false_ids, state, update_token(cache, in_token))
```

Comment:
In this case, I think you can just not return the cache at all.

```suggestion
    to_mlir_if_branch(true, node, on_true, true_ids, state, cache)
    to_mlir_if_branch(false, node, on_false, false_ids, state, cache)
```